xdsl/interpreters/test.py

```diff
@@ -0,0 +1,17 @@
+from xdsl.dialects import test
+from xdsl.interpreter import (
+    Interpreter,
+    InterpreterFunctions,
+    PythonValues,
+    impl,
+    register_impls,
+)
+
+
+@register_impls
+class TestFunctions(InterpreterFunctions):
+    @impl(test.TestOp)
+    def run_test(
+        self, interpreter: Interpreter, op: test.TestOp, args: PythonValues
+    ) -> PythonValues:
+        return (range(len(op.res)),)
```

Comment:
can you please move this to the test directly? Ideally inside the test function. Test ops do not have any runtime semantics, we should not provide implementations for them.
```suggestion
@register_impls
class TestFunctions(InterpreterFunctions):
    @impl(test.TestOp)
    def run_test(
        self, interpreter: Interpreter, op: test.TestOp, args: PythonValues
    ) -> PythonValues:
        return (1,)
```
I think my initial suggestion over Zulip was not good, the idea is to provide a clearly broken function, which happens to be correct when the number of results of the op matches (in this case, 1 result), and test that the interpreter raises an error if it doesn't match (for example, when given a test op with no results).